### PR TITLE
Set up pip-tools

### PR DESCRIPTION
**Pull request by gsganden**
Addresses a dependency issue I encountered and sets up pip-tools to help manage dependencies in the future.

The changes in `learner.py` were reviewed in a previous PR that I made against this branch, so they should not need to be reviewed again.

Pull Request Checklist
 - [ ] All tests in the `tests` folder pass with a local build
 - [x] Pull request includes a description of why we are doing this
 - [ ] Init files import new capabilities to appropriate level of package (if applicable)
 - [x] CHANGELOG has been updated
 - [x] Version in `_version.py` has been updated
 - [ ] README has been updated (if applicable)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,27 +23,15 @@
     install_requires=[
         'fastprogress',
         'joblib',
+        'matplotlib',
```
Comment (gsganden) [inline setup.py:26]: Needed for progress bars in notebooks

File: setup.py [hunk 2]
```diff
@@ -23,27 +23,15 @@
...
-        'transformers>=2.3.0',
+        'transformers>=2.3.0,<4',  # V4 makes breaking changes
         'sentencepiece!=0.1.92',
         'scikit-learn',
         'torch',
         'torchvision==0.2.1',
         'transformers<3.0.0',
         'wildebeest',
     ],
-    extras_requires={
```
Comment (gsganden) [inline setup.py:36]: I suggest we just use the requirements-dev files.

File: octopod/learner.py [hunk 3]
```diff
@@ -88,8 +88,6 @@ def fit(
             PyTorch learning rate scheduler
         step_scheduler_on_batch: bool
             flag of whether to step scheduler on batch (if True) or on epoch (if False)
-        loss_function: function
-            function to calculate loss with in model
```
Comment (gsganden) [inline octopod/learner.py:92]: Unrelated change -- I noticed that this parameter no longer exists.

File: octopod/learner.py [hunk 4]
```diff
@@ -39,7 +39,6 @@ class MultiTaskLearner(object):
         functions. Instead the desired activation functions are applied when needed.
         So we use 'categorical_cross_entropy' and 'bce_logits' loss functions which apply
         softmax and sigmoid activations to their inputs before calculating the loss.
-
```
Comment (gsganden) [inline octopod/learner.py:42]: Aside from the files commented on below, the changes in this file were reviewed in a previous PR that I merged into `pip-tools`, so they should be good to go.

File: setup.py (hunk 2 above)
Comment (sugi-chan) [reply setup.py:36]: yeah just double checked one thing to see if I had put that there initially bc of the read the docs biz, but looks like it is fine. so we should be able to just have this in the `requirements-dev.txt` file

File: octopod/learner.py [hunk 5]
```diff
@@ -99,7 +97,21 @@ def fit(
             The default is `False`, which will keep the final model from the training run.
             `True` will keep the best model from the training run instead of the model
             from the final epoch of the training cycle.
+        smooth_loss_alpha: float
```
Comment (sugi-chan) [inline octopod/learner.py:100]: so default of .2 gives some higher weight to the recent batch but operates similarly to how it was before? a score of like .8 would heavily weigh towards recent... very cool 

(I know I dont have to review this, but its kinda cool to look at)

**Review by sugi-chan — APPROVED** [review]
LGTM, these are cool additions. Always glad to see octopod love.
